neptune/internal/experiment_structure.py

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright (c) 2020, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Dict, Any, Optional, List, TypeVar, Generic
+
+from neptune.exceptions import MetadataInconsistency
+
+T = TypeVar('T')
+
+
+class ExperimentStructure(Generic[T]):
+
+    def __init__(self):
+        self._structure = dict()
+
+    def get_structure(self) -> Dict[str, Any]:
+        return self._structure
+
+    def get(self, path: List[str]) -> Optional[T]:
+        ref = self._structure
+
+        for part in path:
+            if not isinstance(ref, dict):
+                raise MetadataInconsistency("Cannot access path '{}': '{}' is already defined as a variable, "
```

Comment:
So we don't support getting subspaces?

Reply:
I cannot see any reason for supporting it for now. Notice that it is just internal implementation of experiment structure. It's not a `Handler`. We still support accessing namespace via `Handler` object like this: `exp['some/ns']`.